plugins/node/opentelemetry-instrumentation-express/src/types.ts

```diff
@@ -14,15 +14,15 @@
  * limitations under the License.
  */
 
-import type { Request } from 'express';
 import { Span } from '@opentelemetry/api';
 import { InstrumentationConfig } from '@opentelemetry/instrumentation';
 import { ExpressLayerType } from './enums/ExpressLayerType';
 
 export type IgnoreMatcher = string | RegExp | ((name: string) => boolean);
 
-export type ExpressRequestInfo = {
-  request: Request;
+export type ExpressRequestInfo<T = any> = {
```

Comment:
Note for reviewer: type argument `T` defaults to `any` so anyone already using `requestHook` and `spanNameHook` will still work at the type level. The `unknown` type was considered but it will trigger typing errors within these hooks.

Reply:
Theree is a small detail for existing TS users. The type of request property will become of type any. Therefore TS will not provide autocompletion and type checks for it and its sub-properties. This may not be nice to existing users.

To mitigate that we may create our own default type. I've already tried to copy express' request type but there are many links to other types making it a big chunk of code just for typing.

Maybe we could agree on a subset of the request api but this come with the risk of breaking the build of existing TS users of they access a property not defined on this default type.